wolfcrypt/src/port/Espressif/esp32_mp.c

```diff
@@ -330,7 +330,18 @@ static int esp_get_rinv(MATH_INT_T *rinv, MATH_INT_T *M, word32 exp)
 /* Z = X * Y;  */
 int esp_mp_mul(MATH_INT_T* X, MATH_INT_T* Y, MATH_INT_T* Z)
 {
-    int ret = 0;
+    int ret = MP_OKAY; /* assume success until proven wrong */
+
+#ifdef WOLFSSL_SP_INT_NEGATIVE
+    /* neg check: X*Y becomes negative */
+    int neg = (mp_isneg(X) == mp_isneg(Y)) ? MP_ZPOS : MP_NEG; // (X->sign == Y->sign) ? MP_ZPOS : MP_NEG;
+    if (neg) {
```

Comment:
On some compilers this will cause mixed declaration with code. Variable declarations at top of function or brace section.

Reply:
This is unlikely to be a problem on the Espressif environment, but I agree it is a good practice. Fixed.